app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/trustControllers/TrustUtrController.scala

```diff
@@ -0,0 +1,90 @@
+import uk.gov.hmrc.minorentityidentificationfrontend.forms.trustForms.TrustCaptureUtrForm
+import uk.gov.hmrc.minorentityidentificationfrontend.services.{JourneyService, StorageService}
+import uk.gov.hmrc.minorentityidentificationfrontend.views.html.trustViews.capture_utr_trust_page
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class TrustUtrController @Inject()(val authConnector: AuthConnector,
+                                   journeyService: JourneyService,
+                                   storageService: StorageService,
+                                   mcc: MessagesControllerComponents,
+                                   trustView: capture_utr_trust_page
+                                    )(implicit val config: AppConfig,
+                                      executionContext: ExecutionContext) extends FrontendController(mcc) with AuthorisedFunctions {
+
+  def show(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised().retrieve(internalId) {
+        case Some(authInternalId) =>
+          journeyService.getJourneyConfig(journeyId, authInternalId).map {
+            journeyConfig => Ok(trustView(
+                  journeyId = journeyId,
+                  pageConfig = journeyConfig.pageConfig,
+                  formAction = routes.TrustUtrController.submit(journeyId),
+                  form = TrustCaptureUtrForm.trustForm
+                ))
+          }
+        case None =>
+          throw new InternalServerException("Internal ID could not be retrieved from Auth")
+      }
+  }
+
+  def submit(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised().retrieve(internalId) {
+        case Some(authInternalId) =>
+          TrustCaptureUtrForm.trustForm.bindFromRequest().fold(
+            formWithErrors =>
+              journeyService.getJourneyConfig(journeyId, authInternalId).map {
+                journeyConfig => BadRequest(trustView(
+                      journeyId = journeyId,
+                      pageConfig = journeyConfig.pageConfig,
+                      formAction = routes.TrustUtrController.submit(journeyId),
+                      form = formWithErrors
+                    ))
+              },
+            utr =>
+              storageService.storeUtr(journeyId, utr).map {
+                _ => Redirect(routes.CaptureSaPostcodeController.show(journeyId))
+              }
+          )
+        case None => throw new InternalServerException("Internal ID could not be retrieved from Auth")
+      }
+  }
+
+  def noUtr(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised() {
+        storageService.removeUtr(journeyId).map {
+          _ => Redirect(routes.CaptureSaPostcodeController.show(journeyId))
```

Comment:
This is meant to go to the CHRN page so maybe just keep as not implemented for now